packages/react/src/components/dialog/index.tsx

```diff
@@ -42,18 +42,33 @@ export const Dialog = ({
   container,
   dismissable = true,
   modal = true,
-}: Props) => {
+}: DialogProps) => {
+  /**
+   * Check if the container function return value changed every cycle but only
+   * trigger another render when it actually changes.
+   */
+  const containerFuncReturn = (() => {
+    if (typeof container !== "function") return null
+
+    return container()
+  })()
+
+  const __container = useMemo(() => {
+    if (typeof container === "function") return containerFuncReturn
+    return container
+  }, [container, containerFuncReturn])
+
   if (!trigger) return null;
 
   return (
     <RadixDialog.Root modal={modal} open={open} onOpenChange={onOpenChange}>
       <RadixDialog.Trigger asChild>{trigger}</RadixDialog.Trigger>
-      <RadixDialog.Portal container={container}>
+      <RadixDialog.Portal container={__container}>
         {modal && <RadixDialog.Overlay className={styles.overlay} />}
         <RadixDialog.Content
           className={clsx("sid-dialog", styles.wrapper, className)}
           onInteractOutside={(e) => {
-            if (!dismissable) {
+            if (!modal || !dismissable) {
```

Comment:
Thinking about it, is there a need to have both `modal` and `dismissable` as props? I understand this makes the API more flexible, but then the `modal` prop loses its meaning.

I would expect the following behaviour:
- `modal: true`: outside interactions are disabled, the user needs to interact with the dialog, close button is rendered
- `modal: false`: modal is closed on interactions outside, close button is not rendered

Maybe the real question is if there is a case when someone wants to explicitly prohibit any interaction with the app until the consent is resolved?

Reply:
Sorry, when writing this I was convinced I was looking at the `GDPRConsentDialog` and now the lower level dialog component! Agreed with everything you said, thank you!